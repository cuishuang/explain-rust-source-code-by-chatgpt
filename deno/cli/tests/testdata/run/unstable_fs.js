const scope = import.meta.url.slice(-7) === "#worker" ? "worker" : "main";

console.log(scope, Deno.flock);
console.log(scope, Deno.flockSync);
console.log(scope, Deno.funlock);
console.log(scope, Deno.funlockSync);
console.log(scope, Deno.umask);

if (scope === "worker") {
  postMessage("done");
} else {
  const worker = new Worker(`${import.meta.url}#worker`, { type: "module" });
  worker.onmessage = () => Deno.exit(0);
}
