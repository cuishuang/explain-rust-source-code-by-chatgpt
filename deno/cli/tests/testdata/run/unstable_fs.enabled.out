main [AsyncFunction: flock]
main [Function: flockSync]
main [AsyncFunction: funlock]
main [Function: funlockSync]
main [Function: umask]
worker [AsyncFunction: flock]
worker [Function: flockSync]
worker [AsyncFunction: funlock]
worker [Function: funlockSync]
worker [Function: umask]
